File: frontend/src/features/public-form-view/types/question-response-dto.ts
import { AnswerResponseDto } from "./answer-response-dto";
import { QuestionType } from "./question-type";

export interface QuestionResponseDto {
  id: string;
  content: string;
  type: QuestionType;
  imageKey: string | null;
  isRequired: boolean;
  answers: AnswerResponseDto[] | null;
}

File: backend/src/main/java/format/backend/form/dto/QuestionResponseDto.java
package format.backend.form.dto;

import format.backend.form.entity.QuestionType;
import java.util.List;
import org.jspecify.annotations.NonNull;
import org.jspecify.annotations.Nullable;

public record QuestionResponseDto(
        @NonNull String id,

        @NonNull String content,

        @NonNull QuestionType type,

        @Nullable String image,

        @NonNull Boolean isRequired,

        @NonNull List<@NonNull AnswerResponseDto> answers) {}
